src/main/java/com/nexmo/quickstart/insight/AdvancedInsight.java

```diff
@@ -60,11 +61,24 @@ public static void main(String[] args) throws Exception {
         System.out.println("Reachability: " + response.getReachability());
         System.out.println("Ported status: " + response.getPorted());
         RoamingDetails roaming = response.getRoaming();
-        System.out.println("Roaming status: " + roaming.getStatus());
-        if (response.getRoaming().getStatus() == RoamingDetails.RoamingStatus.ROAMING) {
-            System.out.print("    Currently roaming in: " + roaming.getRoamingCountryCode());
-            System.out.println(" on the network " + roaming.getRoamingNetworkName());
+        if (roaming == null) {
+            System.out.println("- No Roaming Info -");
+        } else {
+            System.out.println("Roaming status: " + roaming.getStatus());
+            if (response.getRoaming().getStatus() == RoamingDetails.RoamingStatus.ROAMING) {
+                System.out.print("    Currently roaming in: " + roaming.getRoamingCountryCode());
+                System.out.println(" on the network " + roaming.getRoamingNetworkName());
+            }
         }
 
+        if (CNAM) {
+            System.out.println();
+            System.out.println("CNAM INFO:");
+            System.out.println("Caller Name: " + response.getCallerName());
+            System.out.println("Caller Type: " + response.getCallerType());
+            System.out.println("First, Last: " + response.getFirstName() + ", " + response.getLastName());
```

Comment:
Could this be System.out.println("First name: " + response.getFirstName() + ", Last name: " + response.getLastName());